### DISCUSSION TITLE
Ewok component framework

### DISCUSSION THREAD

**Discussion by Lomacar** [alpinejs/alpine#2497] [2. Show and tell]
Hi all,

For the past several weeks I have been working on a small js library that makes it easy to create and use native web components. I made it in such away that it complements Alpine. I'd love to get some feedback. It could use further testing and refinement. There are a few edge cases that I am still working on, but it works pretty well already.

https://github.com/Lomacar/Ewok

**Comment by danddanddand** [comment]
I only glanced at it but looks like it was fun to code. 

One thing that seemed weird to me was using the name xref and x-ref inside the template. The x- prefix is used to distinguish Alpine stuff (and idea and naming inspired by ref in Vue) so feels like you'd want to use something else for clarity or use your own prefix incase someone tried to pair this on page with Vue/Alpine?

**Reply by Lomacar** [reply]
Thanks for having a look! Actually, the x-ref thing the is one thing I realized I should have worked on more before announcing Ewok. I think it is causing an error still.

But anyway, since Ewok is designed to integrate with Alpine I thought it makes more sense to use the nomenclature of Alpine, rather than invent something else that does the same job. I suppose if you are not using Alpine at all then it may seem out of place, but x-ref looks like cross-reference, so I don't mind the name.

**Reply by Lomacar** [reply]
Hi @sinergantara I don't see how that would work or what advantage that would bring. If you have more thoughts, please elaborate!
